### DISCUSSION TITLE
Is it possible to use different shells inside a single justfile ?

### DISCUSSION THREAD

**Discussion by Roms1383** [casey/just#1659] [Q&A] [answered]
Hi and thanks for `just`, I basically use it everywhere and absolutely love it.
There's only one thing which I struggle with : in some project, I sometimes have to use different shells, for example `powershell` during development, but `bash` inside Github CI. I've tried different approaches, but none is as ergonomic as I initially hoped.

So long question short : is it possible to use e.g. `powershell` and `bash` inside the same project, without having to delegate to `justfile` in sub-directories or having to specify `--shell` / `--shell-arg` on each command call ? Maybe it's easy and I just missed something.

**Comment by casey** [comment, marked as answer]
It isn't possible to have different values of the `shell` setting, but it is possible to use shebang recipes to use a different interpreter:

```
foo:
  #!/usr/bin/env bash
  echo hello

bar:
  #!/usr/bin/env python3
  print('hello')
```

**Reply by Roms1383** [reply]
thanks @casey
